src/dmd/backend/cod3.d

```diff
@@ -4095,8 +4095,11 @@ void prolog_loadparams(ref CodeBuilder cdb, tym_t tyf, bool pushalloc, out regm_
         {
             shadowregm |= mask(preg);
             opcode_t op = 0x89;                  // MOV x[EBP],preg
+            type* ta = t;
+            while ((ta.Tty & TYarray) && ta.Tnext)
+                ta = ta.Tnext;
             if (isXMMreg(preg))
-                op = xmmstore((t.Tty & TYarray) && t.Tnext ? t.Tnext.Tty : t.Tty);
+                op = xmmstore(ta.Tty);
```

Comment:
I suggest the following:
```
static type* type_arrayBase(type* ta)
{
    while (tybasic(ta.Tty) == TYarray)
        ta = ta.Tnext;
    return ta;
}

const opcode_t op = isXMMreg(preg)
        ? xmmstore(type_arrayBase(t).Tty)
        : 0x89;    // MOV x[EBP],preg
```
and in a followon refactoring PR, move type_arrayBase() to dtype.d, and replace other loops of this sort (there's one in cod1.d) with a call to this function.

Note that this achieves a single assignment to `op` and so it can be `const`. It also fixes the `&` bug, which should be `==`.

Reply:
Another reason for this change is the loop only gets executed if it's an XMM reg, instead of always.

Reply:
yes there's no disagreement on fact that this is better.